### PR TITLE
agent: Mark --lb feature deprecated for removal in 1.7

### LINKED ISSUE #8599: deprecate bpf_lb.c to schedule for removal in 1.7
Deprecate stand-alone bpf_lb.c so we can remove it fully in 1.7.

### PR DESCRIPTION
**Pull request by tgraf**
Fixes: #8599

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8786)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

File: Documentation/install/upgrade.rst [hunk 1]
```diff
@@ -331,6 +331,11 @@ Deprecated options
   or later to 1.4.x may result in disruption of connections that connect via
   services.
 
+* ``lb``: The ``--lb`` feature has been deprecated. It has not been in use and
+  was well tested. If you need load-balancing with pickup from a device
```
Comment (brb) [inline Documentation/install/upgrade.rst]: Nits:
1. s/was well tested/has not been well tested/
2. s/If you need load-balancing with pickup from a device directly/If you need load-balancing on a particular device/

**Comment by ianvernon** [comment]
@tgraf is this ready for review?

**Review by aanm — APPROVED** [review]

**Comment by tgraf** [comment]
test-docs-please

**Review by borkmann — APPROVED** [review]

**Pull request merged**